docs/azure/integrations-overview.md

```diff
@@ -0,0 +1,196 @@
+
+### Azure provisioning customization
+
+All .NET Aspire Azure hosting integrations expose various Azure resources, and they're all subclasses of the <xref:Aspire.Hosting.Azure.AzureProvisioningResource> type—which itself inherits the <xref:Aspire.Hosting.Azure.AzureBicepResource>. This enables extensions that are generically type-constrained to this type, allowing for a fluent API to customize the infrastructure to your liking. While the Azure SDK relies on defaults, you're free to influence the generated Bicep using these APIs.
+
+#### Configure infrastructure
+
+Regardless of the Azure resource you're working with, to configure its underlying infrastructure, you chain a call to the <xref:Aspire.Hosting.AzureProvisioningResourceExtensions.ConfigureInfrastructure*> extension method. This method allows you to customize the infrastructure of the Azure resource by passing a `configure` delegate of type `Action<AzureResourceInfrastructure>`. The <xref:Aspire.Hosting.Azure.AzureResourceInfrastructure> type is a subclass of the <xref:Azure.Provisioning.Infrastructure?displayProperty=fullName>. This type exposes a massive API surface area for configuring the underlying infrastructure of the Azure resource.
+
+Each `configure` action:
+
+- Calls the <xref:Azure.Provisioning.Infrastructure.GetProvisionableResources> API to get the provisionable resources.
+- Filters the provisionable resources based on target resource type to customize.
+- Applies the desired configuration to the resource.
+
+Consider the following example:
+
+:::code language="csharp" source="../snippets/azure/AppHost/Program.ConfigureInfrastructure.cs" id="infra":::
+
+The preceding code:
+
+- Adds a parameter named `storage-sku`.
+- Adds Azure Storage with the <xref:Aspire.Hosting.AzureStorageExtensions.AddAzureStorage*> API named `storage`.
+- Chains a call to `ConfigureInfrastructure` to customize the Azure Storage infrastructure:
+  - Gets the provisionable resources.
+  - Filters to a single <xref:Azure.Provisioning.Storage.StorageAccount>.
+  - Assigns the `storage-sku` parameter to the <xref:Azure.Provisioning.Storage.StorageAccount.Sku?displayProperty=nameWithType> property:
+    - A new instance of the <xref:Azure.Provisioning.Storage.StorageSku> has its `Name` property assigned from the result of the <xref:Aspire.Hosting.AzureProvisioningResourceExtensions.AsProvisioningParameter*> API.
+
+This exemplifies flowing an [external parameter](../fundamentals/external-parameters.md) into the Azure Storage infrastructure, resulting in the generated Bicep file reflecting the desired configuration.
```

Comment:
I'm not sure if we already have it, but it might be nice to explain where the parameter value comes from - or link off to parameters.

Reply:
That's what the external parameter link is for, it's detailed in there.